### DISCUSSION TITLE
Multiple inlineButton pressing = multiple callback's

### DISCUSSION THREAD

**Discussion by jrgektor** [aiogram/aiogram#1499] [Q&A]
I ask for help with the following question.

Here's an example of a situation:
User presses an inline keyboard button, and a callback is sent to the server.
The function starts working, the user can press the same button several more times. And each of it will sent new cannlack query to server. And these additional calls will also be processed when first callback function finished.

Is there any way to avoid this?
I even tried to remove all buttons with callbacks immediately after clicking on the inline button, but the user still has the opportunity to click on the button several times in a few milliseconds and send several callback requests. But I need the request to come only in a single copy.

**Comment by JrooTJunior** [comment]
First of all you need to enable [events isolation on the Dispatcher level](https://docs.aiogram.dev/en/dev-3.x/dispatcher/dispatcher.html#aiogram.dispatcher.dispatcher.Dispatcher) to avoid simultaneous processing of events for one user.

After that you can implement rate-limiter for user requests. It can be implemented in a different ways, just google how to do it.
